io/io/src/TGenCollectionProxy.cxx

```diff
@@ -999,9 +1003,10 @@ void TGenCollectionProxy::UpdateValueClass(const TClass *oldValueType, TClass *n
    // Note that we do not need to update anything if we have not yet been
    // initialized.  In addition (see ROOT-6040) doing an initialization here
    // might hence a nested dlopen (due to autoloading).
-   if (fValue.load() && (*fValue).fType == oldValueType) {
+   auto value = fValue.load(std::memory_order_relaxed);
+   if (value && (*value).fType == oldValueType) {
       // Set pointer to the TClass representing the content.
-      (*fValue).fType = newValueType;
+      (*value).fType = newValueType;
```

Comment:
Changing `fType` does not appear to be done in a thread safe manner. Is that going to be a problem?

Reply:
I agree that it was already the case in the current code and did not feel the change here made the current situation any worse.

Reply:
The good news is that code was actually render moot by a (slightly) later commit.
See
   8cfa620611777bb56562dc4eb984f2a947263ca6
which brings this code and
   c9460ae556c84c9a7febb7cef6865e8e04c2621a
which makes it moot (essentially already replacing with some additional indirection the storage with an atomic)